waves/src/Bobtimus.tsx

```diff
@@ -135,9 +155,14 @@ export async function postLoanFinalization(txHex: string) {
         body: JSON.stringify({ tx_hex: txHex }),
     });
 
-    if (res.status !== 200) {
+    if (res.status === 400) {
```

Comment:
To be fully correct, this condition isn't quite enough.

1. The response code 400 tells you that something went wrong but it doesn't tell you anything about the contents body.
2. To learn about what is in the body, we would need to look at the `Content-Type` header. A content-type of `application/problem+json` would tell us that our body has the shape of a _problem_.
3. Knowing that we are dealing with a _problem_, we can probe for the [_type_](https://datatracker.ietf.org/doc/html/rfc7807#section-3.1) of problem. In our case, we would want to check for a type that is something like `/errors/validation/loan`. As per the spec, this could be a URI that actually points to more documentation about this error.

As a result, we would actually want to check something like (pseudo-code):

```ts
if (res.status == 400 && res.contentType == 'application/problem+json') {
	const problem = await res.json();

	if (problem.type == '/errors/validation/loan') {
		// do specific error handling for loan validation errors
	}
}
```

You may ask "what is the point of this?" and the answer is: If we do this just _here_, there isn't really a point. However, what we can/could do in the future is have a generic HTTP middleware component that probs for these response kinds and automatically throws the appropriate error based on the details of the problem.

On a broader context: The idea here is that we don't rely on out-of-band information for how to treat a certain response but only make decisions based on the self-descriptive nature of the response.

It would also allow us to have a generic error handler that displays a toast with the detail anytime we receive a problem that we don't actually know about.